src/main_functions.c

```diff
@@ -0,0 +1,530 @@
+	while (tkn != NULL) {
+
+		if (valid_word(tkn, dict) == EXIT_FAILURE){
+			//printf("%s ", tkn);
+			underline_misspelled(tkn, underline);
+			add_to_misspelled(tkn, misspelled);
+		}
+		else if (valid_word(tkn, dict) == EXIT_SUCCESS) {
+			//printf("%s ", tkn);
+			underline_correct_spelling(tkn, underline);
+		}
+		else {
+			printf("error processing text");
+			return EXIT_FAILURE;
+		}
+		tkn = strtok(NULL, " ,.-");
+	}
+	return EXIT_SUCCESS;
+
+}
+
+//reference from https://stackoverflow.com/questions/32413667/replace-all-occurrences-of-a-substring-in-a-string-in-c
+char* correct_line(char* line, char* old_word, char* new_word)
+{
+	char buffer[2000] = {0}; //again, we might need to modify our size estimates
+	char *insert_point = &buffer[0];
+	char *tmp = line;
+	int old_length = strlen(old_word);
+	int new_length = strlen(new_word);
+
+	while(1) {
+		char *p = strstr(tmp, old_word);
+
+		if (p == NULL) {
+			strcpy(insert_point, tmp);
+			break;
+		}
+
+		memcpy(insert_point, tmp, p - tmp);
+		insert_point += p - tmp;
+
+		memcpy(insert_point, new_word, new_length);
+		insert_point += new_length;
+
+		tmp = p + old_length;
+	}
+
+	strcpy(line, buffer);
+
+	return line;
+
+}
+
+
+
+int initialize_misspelled(char **misspelled, int length)
+{
+
+	if (misspelled==NULL) {
+		return EXIT_FAILURE;
+	}
+    int i = 0;
+	for ( ; i < length; i++) {
+    	misspelled[i] = NULL; //initialize each element to be NULL
+    }
+
+    return EXIT_SUCCESS;
+}
+
+
+
+/*
+ *	III. Interactive Mode
+ */
+
+char* edit_interactive(char* line, dict_t* dict, int linenumber)
+{
+    char *line_copy = malloc(strlen(line));
+    strcpy(line_copy, line); //maintain a copy of the line to preserve original line: line will be parsed into individual words
+    
+    char *line_copy2 = malloc(strlen(line));
+    strcpy(line_copy2, line); 
+    int max_no_suggestions = 2; //should the user decide this?
+
+
+    int length = strlen(line) / 3; //approximate 3 chars per word to be safe
+    char *misspelled[length]; //generates an empty array where the misspelled words in a line will be stored
+    initialize_misspelled(misspelled, length);
+
+    char *underline = (char *)malloc(strlen(line + 1)); //generate an empty array where the underline will go
+    underline[0] = '\0'; 
+    //char underline[strlen(line)] = "";
+
+
+    parse_string(line, dict, underline, misspelled); //identify misspelled words and add to bad_word, 
+    //add to underline function 
+    
+    printf("Current line number is %d: \n", linenumber);
+    printf("%s", line_copy);
+    printf("\n");
+    printf("%s", underline);
+    printf("\n\n");
+
+    // printf("Misspelled words in this sentence are: ");
+
+    // int i = 0;
+    // while(badwords[i] != NULL) {
+    // 	printf("%d: %s ",i+1, badwords[i]);
+    // 	i++;
+    // }
+    // printf("\n");
+    // int number1;
+    // printf("Enter the number value of the word you would like to change: ");
+    // scanf("%d", &number1);
```

Comment:
Remove commented out code and print statements

Reply:
done